FrontBundle/Routing/Database/OpenOrchestraDatabaseRouter.php

```diff
@@ -0,0 +1,178 @@
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Class OpenOrchestraDatabaseRouter
+ */
+class OpenOrchestraDatabaseRouter implements RouterInterface
+{
+    /**
+     * @var RequestContext
+     */
+    protected $context;
+
+    protected $routeDocumentCollectionToRouteCollectionTransformer;
+    protected $routeDocumentToValueObjectTransformer;
+    protected $routeDocumentRepository;
+    protected $options = array();
+    protected $routeCollection;
+    protected $requestStack;
+    protected $nodeManager;
+    protected $generator;
+    protected $matcher;
+
+    /**
+     * @param RouteDocumentRepositoryInterface                    $routeDocumentRepository
+     * @param RouteDocumentToValueObjectTransformer               $routeDocumentToValueObjectTransformer
+     * @param RouteDocumentCollectionToRouteCollectionTransformer $routeDocumentCollectionToRouteCollectionTransformer
+     * @param RequestStack                                        $requestStack
+     * @param NodeManager                                         $nodeManager
+     * @param array                                               $options
+     */
+    public function __construct(
+        RouteDocumentRepositoryInterface $routeDocumentRepository,
+        RouteDocumentToValueObjectTransformer $routeDocumentToValueObjectTransformer,
+        RouteDocumentCollectionToRouteCollectionTransformer $routeDocumentCollectionToRouteCollectionTransformer,
+        RequestStack $requestStack,
+        NodeManager $nodeManager,
+        array $options = array()
+    )
+    {
+        $this->routeDocumentCollectionToRouteCollectionTransformer = $routeDocumentCollectionToRouteCollectionTransformer;
+        $this->routeDocumentToValueObjectTransformer = $routeDocumentToValueObjectTransformer;
+        $this->routeDocumentRepository = $routeDocumentRepository;
+        $this->requestStack = $requestStack;
+        $this->nodeManager = $nodeManager;
+
+        $this->options = array_merge(array(
+            'generator_class' => 'OpenOrchestra\FrontBundle\Routing\Database\OpenOrchestraDatabaseUrlGenerator',
+            'matcher_class' => 'OpenOrchestra\FrontBundle\Routing\Database\OpenOrchestraDatabaseUrlMatcher'
+        ), $options);
+    }
+
+    /**
+     * Sets the request context.
+     *
+     * @param RequestContext $context The context
+     */
+    public function setContext(RequestContext $context)
+    {
+        if (null !== $this->generator) {
+            $this->getGenerator()->setContext($context);
+        }
+        $this->context = $context;
+    }
+
+    /**
+     * Gets the request context.
+     *
+     * @return RequestContext The context
+     */
+    public function getContext()
+    {
+        return $this->context;
+    }
+
+    /**
+     * Gets the RouteCollection instance associated with this Router.
+     *
+     * @return RouteCollection A RouteCollection instance
+     */
+    public function getRouteCollection()
```

Comment:
Why hydrate all the routes ? It will be far less performant than the ancient router. I don't see where this is used and why.